packets.go

```diff
@@ -380,20 +422,39 @@ func (mc *mysqlConn) writeHandshakeResponsePacket(authResp []byte, plugin string
 	data[pos] = 0x00
 	pos++
 
-	// Auth Data [length encoded integer]
-	pos += copy(data[pos:], authRespLEI)
+	// Auth Data [length encoded integer + data] if clientPluginAuthLenEncClientData
+	// clientSecureConn => 1 byte len + data
+	// else null-terminated
+	if clientFlags&clientPluginAuthLenEncClientData != 0 {
+		pos += copy(data[pos:], authRespLEI)
+	} else if clientFlags&clientSecureConn != 0 {
+		data[pos] = uint8(len(authResp))
+		pos++
+	}
```

Comment:
Why is this needed?

Reply:
@andygrunwald – do you remember any context on this one? I tried removing this block and confirmed many tests fail. I'll dig in deeper tomorrow, but wanted to start off seeing if you remembered this code.